src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/readthreading/ReadThreadingAssembler.java

```diff
@@ -129,49 +128,65 @@ public AssemblyResultSet runLocalAssembly(final AssemblyRegion assemblyRegion,
             correctedReads = assemblyRegion.getReads();
         }
 
-        final List<SeqGraph> nonRefGraphs = new LinkedList<>();
+        final List<ReadThreadingGraph> nonRefRTGraphs = new LinkedList<>();
+        final List<SeqGraph> nonRefSeqGraphs = new LinkedList<>();
         final AssemblyResultSet resultSet = new AssemblyResultSet();
         resultSet.setRegionForGenotyping(assemblyRegion);
         resultSet.setFullReferenceWithPadding(fullReferenceWithPadding);
         resultSet.setPaddedReferenceLoc(refLoc);
         final SimpleInterval activeRegionExtendedLocation = assemblyRegion.getExtendedSpan();
         refHaplotype.setGenomeLocation(activeRegionExtendedLocation);
         resultSet.add(refHaplotype);
-        final Map<SeqGraph,AssemblyResult> assemblyResultByGraph = new HashMap<>();
+        final Map<ReadThreadingGraph,AssemblyResult> assemblyResultByRTGraph = new HashMap<>();
+        final Map<SeqGraph,AssemblyResult> assemblyResultBySeqGraph = new HashMap<>();
         // create the graphs by calling our subclass assemble method
         for ( final AssemblyResult result : assemble(correctedReads, refHaplotype, header, aligner) ) {
             if ( result.getStatus() == AssemblyResult.Status.ASSEMBLED_SOME_VARIATION ) {
                 // do some QC on the graph
-                sanityCheckGraph(result.getGraph(), refHaplotype);
-                // add it to graphs with meaningful non-reference features
-                assemblyResultByGraph.put(result.getGraph(),result);
-                nonRefGraphs.add(result.getGraph());
+                if (generateSeqGraph) {
+                    sanityCheckGraph(result.getSeqGraph(), refHaplotype);
+                    // add it to graphs with meaningful non-reference features
+                    assemblyResultBySeqGraph.put(result.getSeqGraph(),result);
+                    nonRefSeqGraphs.add(result.getSeqGraph());
+                } else {
+                    sanityCheckGraph(result.getThreadingGraph(), refHaplotype);
+                    // add it to graphs with meaningful non-reference features
+                    assemblyResultByRTGraph.put(result.getThreadingGraph(),result);
+                    nonRefRTGraphs.add(result.getThreadingGraph());
+                }
+
             }
         }
 
         // add assembled alt haplotypes to the {@code resultSet}
-        findBestPaths(nonRefGraphs, refHaplotype, refLoc, activeRegionExtendedLocation, assemblyResultByGraph, resultSet, aligner);
+        if (generateSeqGraph) {
```

Comment:
Can you write this as `findBestPaths(generateSeqGraph ? nonRefSeqGraphs : nonRefRTGRaphs, refHaplotype. . .)`

Reply:
Unfortunately not because of the way I am using generics. The first and fifth inputs are bound by the same equality constraint which the compiler complains about.